static/src/javascripts/projects/common/modules/component.spec.js

```diff
@@ -0,0 +1,426 @@
+// @flow
+
+import fetchJSON_ from 'lib/fetch-json';
+
+import { Component } from './component';
+
+const fetchJSON: JestMockFn<*, *> = (fetchJSON_: any);
+
+const mockResponse = {
+    html: '<p>html</p>',
+    other: '<p>other</p>',
+};
+
+const createComponent = (props?: Object = {}): Component => {
+    const component = new Component();
+    const defaults = {
+        endpoint: 'whatever',
+    };
+
+    Object.assign(component, defaults, props);
+
+    return component;
+};
+
+jest.mock('lib/fetch-json', () => jest.fn(() => Promise.resolve(mockResponse)));
```

Comment:
Now, that's an interesting workaround to the `mockResponse is not defined` problem 😲

Reply:
How do you mean? I thought I'm doing the trick here

```js
 const fetchJSON: JestMockFn<*, *> = (fetchJSON_: any); 
```

😆

Reply:
This is solving a different problem. For instance, if you mocked a function that returned a variable defined in the module scope:

```js
const blah = 'foo'
jest.mock('whatever', () => {
  return blah
})
```

Jest would implicitly hoist the mocked function above the `const` declaration into the temporal dead zone, throwing a type error. For some reason adding that extra `jest.fn(...)` wrapper gets around this.